plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/NewProjectWizardPage.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.wizard.IWizard;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.jboss.tools.common.ui.databinding.ParametrizableWizardPageSupport;
+import org.jboss.tools.common.ui.databinding.ValueBindingBuilder;
+import org.jboss.tools.openshift.internal.common.ui.databinding.RequiredControlDecorationUpdater;
+import org.jboss.tools.openshift.internal.common.ui.wizard.AbstractOpenShiftWizardPage;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIMessages;
+import org.jboss.tools.openshift.internal.ui.validator.ProjectNameValidator;
+
+/**
+ * @author jeff.cantrill
+ */
+public class NewProjectWizardPage extends AbstractOpenShiftWizardPage {
+
+	private NewProjectWizardModel model;
+
+	public NewProjectWizardPage(NewProjectWizardModel model, IWizard wizard) {
+		this("New OpenShift Project", "Please enter the project information", model, wizard);
+	}
+
+	protected NewProjectWizardPage(String title, String description, NewProjectWizardModel model, IWizard wizard) {
+		super(title, description, "", wizard);
+		this.model = model;
+	}
+	
+	protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+		GridDataFactory.fillDefaults().grab(true, false).align(SWT.FILL, SWT.TOP).applyTo(parent);
+		GridLayoutFactory.fillDefaults().margins(6, 6).numColumns(2).applyTo(parent);
+
+		// project name
+		Label lblName = new Label(parent, SWT.NONE);
+		lblName.setText(OpenShiftUIMessages.Name);
+		GridDataFactory.fillDefaults()
+				.align(SWT.LEFT, SWT.CENTER).applyTo(lblName);
+		
+		Text txtName = new Text(parent, SWT.BORDER);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.CENTER).grab(true, false).applyTo(txtName);
+		IObservableValue nameTextObservable = WidgetProperties.text(SWT.Modify).observe(txtName);
+		Binding nameBinding = ValueBindingBuilder
+				.bind(nameTextObservable)
+				.validatingAfterConvert(new ProjectNameValidator(getDescription()))
+				.to(BeanProperties.value(NewProjectWizardModel.PROPERTY_PROJECT_NAME).observe(model))
+				.in(dbc);
+		ControlDecorationSupport.create(nameBinding, SWT.LEFT | SWT.TOP, null,
+				new RequiredControlDecorationUpdater());
+
+		// display name
+		Label lblDisplayName = new Label(parent, SWT.NONE);
+		lblDisplayName.setText(OpenShiftUIMessages.DisplayName);
+		GridDataFactory.fillDefaults()
+			.align(SWT.LEFT, SWT.CENTER).applyTo(lblDisplayName);
+		
+		Text txtDispalayName = new Text(parent, SWT.BORDER);
+		GridDataFactory.fillDefaults()
+			.align(SWT.FILL, SWT.CENTER).grab(true, false).applyTo(txtDispalayName);
+		ValueBindingBuilder
+			.bind(WidgetProperties.text(SWT.Modify).observe(txtDispalayName))
+			.validatingAfterConvert(new IValidator() {
+
+				@Override
+				public IStatus validate(Object value) {
+					String param = (String) value;
+					if(StringUtils.isNotEmpty(param)) {
```

Comment:
is displayName not required? we're returning ok() if it's empty, is this correct?